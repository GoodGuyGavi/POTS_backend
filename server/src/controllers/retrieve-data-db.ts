import { IDBModel } from '../commons/types';

const getAllDataDB = (db: IDBModel<any>) => async () => {
	return db.getAll();
};
const getByIDDB = (db: IDBModel<any>) => async id => {
	return db.getById(id);
};

// const getByDeliveryIdDB = (db: IDBModel<any>) => async id => {
//   return db.getByDeliverId(id);
// };

// const getAllByDriverDB = (db: IDBModel<any>) => async id => {
//   return db.getAllByDriver(id);
// };

export { getAllDataDB, getByIDDB };
