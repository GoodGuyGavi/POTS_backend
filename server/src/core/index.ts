import { createUCCreateSupplier } from './_usecases';

export { createUCCreateSupplier };
