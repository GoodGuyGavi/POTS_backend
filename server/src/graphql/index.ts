import { mergeTypes } from 'merge-graphql-schemas';

// import address from "./address";

const typeDefs = [
	//   address,
];

// NOTE: 2nd param is optional, and defaults to false
// Only use if you have defined the same type multiple times in
// different files and wish to attempt merging them together.
export default mergeTypes(typeDefs, { all: true });
