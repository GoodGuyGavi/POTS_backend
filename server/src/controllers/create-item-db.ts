import { IDBModel } from '../commons/types';
import { createUCCreateItem } from '../core/';

const createCreateItemDB = (db: IDBModel<any>) => async i => {
	const createItem = createUCCreateItem();
	const newItem = createItem(i);
	console.log('contorller', newItem);
	return db.insert(newItem);
};

export { createCreateItemDB };
