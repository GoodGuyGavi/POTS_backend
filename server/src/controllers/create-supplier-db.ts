import { IDBModel } from '../commons/types';
import { createUCCreateSupplier } from '../core/_usecases';

const createCreateSupplierDB = (db: IDBModel<any>) => async u => {
	const createSupplier = createUCCreateSupplier();
	const newSupplier = createSupplier(u);
	return db.insert(newSupplier);
};

export { createCreateSupplierDB };
