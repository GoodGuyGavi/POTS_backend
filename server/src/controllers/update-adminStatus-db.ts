import { IDBModel } from '../commons/types';
import { updateUCAdminStatus } from '../core/_usecases';

const updateAdminStatusByIDDB = (db: IDBModel<any>) => async data => {
  const a = await db.getById(data.id);
  const updateAdminStatus = updateUCAdminStatus();
  const newPurchaseOrder = updateAdminStatus(data, a);
  return db.updateById({ ...newPurchaseOrder, id: data.id });
};

export { updateAdminStatusByIDDB };
