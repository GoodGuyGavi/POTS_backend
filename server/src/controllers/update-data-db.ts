import { IDBModel } from '../commons/types';
import { updateUCUser } from '../core/_usecases';

const UpdateByIDDB = (db: IDBModel<any>) => async data => {
	return db.updateById(data);
};

export { UpdateByIDDB };
