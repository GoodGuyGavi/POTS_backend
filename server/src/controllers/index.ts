//User
import { createCreateUserDB } from './create-user-db';
import { DeleteRecordByIDDB } from './delete-data-db';
import { updateUserByIDDB } from './update-user-db';

//Supplier
import { createCreateSupplierDB } from './create-supplier-db';

import { getAllDataDB, getByIDDB } from './retrieve-data-db';

export { getAllDataDB, getByIDDB, createCreateUserDB, DeleteRecordByIDDB, updateUserByIDDB, createCreateSupplierDB };
