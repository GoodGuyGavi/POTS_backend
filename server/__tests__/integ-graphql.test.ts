import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import mockData from './mocks/mock-data';
import * as controllers from '../src/controllers';
import { constructTestServer } from './__utils';
const {
	//Generic
	getAllDataDB,
	getByIDDB,
	DeleteRecordByIDDB,
	//User
	createCreateUserDB,
	updateUserByIDDB,
	//Supplie
	createCreateSupplierDB,
	updateSupplierByIDDB,
} = controllers;

//User

const userMock = {
	insert: jest.fn(async input => {
		return { id: '1', ...input };
	}),
	getById: jest.fn(async id => {
		const filterData = data => {
			if (data.id === id) {
				return data;
			}
		};
		const res = mockData.users.filter(filterData);

		return res[0] || null;
	}),
	getAll: jest.fn(async () => {
		return mockData.users;
	}),
	updateById: jest.fn(async input => {
		return { ...input };
	}),
	deleteById: async id => {
		const filterData = data => {
			if (data.id === id) {
				return data;
			}
		};
		const res = mockData.users.filter(filterData);
		return res[0] || null;
	},
};

//Supplier

const supplierMock = {
	insert: jest.fn(async input => {
		return { id: '1', ...input };
	}),
	getAll: jest.fn(async () => {
		return mockData.suppliers;
	}),
	getById: jest.fn(async id => {
		const filterData = cust => {
			if (cust.id === id) {
				return cust;
			}
		};

		const res = mockData.suppliers.filter(filterData);
		return res[0] || null;
	}),

	deleteById: async id => {
		const filterData = cust => {
			if (cust.id === id) {
				return cust;
			}
		};

		const res = mockData.suppliers.filter(filterData);
		return res[0] || null;
	},

	updateById: jest.fn(async input => {
		return { ...input };
	}),
};

const { server }: any = constructTestServer({
	context: {
		//User
		createUser: createCreateUserDB(userMock),
		getUserById: getByIDDB(userMock),
		getAllUser: getAllDataDB(userMock),
		deleteUserById: DeleteRecordByIDDB(userMock),
		updateUserById: updateUserByIDDB(userMock),
		//Supplier
		createSupplier: createCreateSupplierDB(supplierMock),
		getSupplierById: getByIDDB(supplierMock),
		getAllSuppliers: getAllDataDB(supplierMock),
		deleteSupplierById: DeleteRecordByIDDB(supplierMock),
		updateSupplierById: updateSupplierByIDDB(supplierMock),
	},
});

describe('Queries', () => {
	//User Queries
	it('should fetch all user', async () => {
		const USER_ALL = gql`
			query {
				allUsers {
					id
					name
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({ query: USER_ALL });

		expect(res).toMatchSnapshot();
	});

	it('should fetch one user', async () => {
		const SINGLE_USER = gql`
			query u($id: String!) {
				user(id: $id) {
					name
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_USER,
			variables: { id: '1' },
		});

		expect(res).toMatchSnapshot();
	});

	it('should error when no user', async () => {
		const SINGLE_USER = gql`
			query u($id: String!) {
				user(id: $id) {
					name
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_USER,
			variables: { id: '' },
		});

		expect(res).toMatchSnapshot();
	});

	//Supplier Queries
	it('should fetch all suppliers', async () => {
		const SUPPLIER_ALL = gql`
			query {
				allSuppliers {
					name
					address {
						id
					}
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({ query: SUPPLIER_ALL });

		expect(res).toMatchSnapshot();
	});

	it('should fetch one supplier', async () => {
		const SINGLE_SUPPLIER = gql`
			query supp($id: ID!) {
				supplier(id: $id) {
					name
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_SUPPLIER,
			variables: { id: '1' },
		});

		expect(res).toMatchSnapshot();
	});

	it('should error when no supplier', async () => {
		const SINGLE_SUPPLIER = gql`
			query supp($id: ID!) {
				supplier(id: $id) {
					name
				}
			}
		`;

		const { query } = createTestClient(server);
		const res = await query({
			query: SINGLE_SUPPLIER,
			variables: { id: '' },
		});

		expect(res).toMatchSnapshot();
	});

	//Mutations

	//User Mutations
	it('create a user', async () => {
		const CREATE_USER = gql`
			mutation createUser($name: String) {
				createUser(name: $name) {
					id
					name
				}
			}
		`;

		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: CREATE_USER,
			variables: {
				name: 'User 1',
			},
		});

		expect(res.errors).toBeUndefined();
		expect(userMock.insert.mock.calls.length).toBe(1);
		expect(res.data).toMatchObject({
			createUser: {
				id: '1',
				name: 'User 1',
			},
		});
		expect(res).toMatchSnapshot();
	});

	it('delete a user', async () => {
		const DELETE_USER = gql`
			mutation u($id: String!) {
				deleteUser(id: $id) {
					id
					name
				}
			}
		`;
		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: DELETE_USER,
			variables: { id: 'U1' },
		});
		expect(res.errors).toBeUndefined();
		expect(res).toMatchSnapshot();
	});

	it('update a user', async () => {
		const UPDATE_USER = gql`
			mutation u($name: String!) {
				updateUser(name: $name) {
					name
				}
			}
		`;
		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: UPDATE_USER,
			variables: {
				id: '1',
				name: 'User 1',
			},
		});
		expect(res.errors).toBeUndefined();
		expect(res).toMatchSnapshot();
	});

	//Supplier Mutation
	it('create a supplier', async () => {
		const CREATE_SUPPLIER = gql`
			mutation createSupp($supplier: SupplierInput!) {
				createSupplier(supplier: $supplier) {
					id
					name
				}
			}
		`;

		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: CREATE_SUPPLIER,
			variables: {
				supplier: {
					name: 'The Supplier',
					address: {
						building_name: 'bu',
						street: 'St',
						city: 'cty',
						state: 'stat',
						zip_code: 'zip',
					},
				},
			},
		});

		expect(res.errors).toBeUndefined();
		expect(supplierMock.insert.mock.calls.length).toBe(1);
		expect(res.data).toMatchObject({
			createSupplier: {
				id: '1',
				name: 'The Supplier',
			},
		});
		expect(res).toMatchSnapshot();
	});

	it('update a supplier', async () => {
		const UPDATE_SUPPLIER = gql`
			mutation supp($supplier: UpdateSupplierInput!) {
				updateSupplier(supplier: $supplier) {
					id
					name
					address {
						building_name
						street
					}
				}
			}
		`;

		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: UPDATE_SUPPLIER,
			variables: {
				customer: {
					id: '5dae933089d8fe07b8c6da18',
					name: 'New name for this one',
					address: 'A1',
				},
			},
		});

		expect(res).toMatchSnapshot();
	});

	it('delete a supplier', async () => {
		const DELETE_SUPPLIER = gql`
			mutation supp($id: ID!) {
				deleteSupplier(id: $id) {
					name
				}
			}
		`;

		const { mutate } = createTestClient(server);
		const res = await mutate({
			mutation: DELETE_SUPPLIER,
			variables: { id: 'S1' },
		});

		expect(res).toMatchSnapshot();
	});
});
