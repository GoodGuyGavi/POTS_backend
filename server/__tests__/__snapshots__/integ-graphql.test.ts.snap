// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries create a supplier 1`] = `
Object {
  "data": Object {
    "createSupplier": Object {
      "contactNumber": "1234567",
      "contactPerson": "Basil Valdez",
      "id": "1",
      "supplierName": "The Supplier",
      "supplierNo": "001",
      "tin": "12345",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries create a user 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "id": "1",
      "password": "12345",
      "userLevel": "Admin",
      "userName": "User Name1",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries delete a supplier 1`] = `
Object {
  "data": Object {
    "deleteSupplier": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries delete a user 1`] = `
Object {
  "data": Object {
    "deleteUser": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no address 1`] = `
Object {
  "data": Object {
    "address": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no item 1`] = `
Object {
  "data": Object {
    "item": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no purchase order 1`] = `
Object {
  "data": Object {
    "purchaseOrder": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no schedule line 1`] = `
Object {
  "data": Object {
    "scheduleLine": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no supplier 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [ValidationError: Cannot query field "supplier" on type "Supplier". Did you mean "supplierNo" or "supplierName"?],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no user 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should error when no user 2`] = `
Object {
  "data": Object {
    "supplierStatus": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all address 1`] = `
Object {
  "data": Object {
    "allAddresss": Array [
      Object {
        "building_name": "building name",
        "city": "city",
        "id": "A1",
        "state": "state",
        "street": "street",
        "zip_code": "zip_code",
      },
      Object {
        "building_name": "building name 2",
        "city": "city 2",
        "id": "A2",
        "state": "state 2",
        "street": "street 2",
        "zip_code": "zip_code 2",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all items 1`] = `
Object {
  "data": Object {
    "allItems": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field Item.id.],
    [GraphQLError: Cannot return null for non-nullable field Item.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all purchase orders 1`] = `
Object {
  "data": Object {
    "allPurchaseOrders": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field PurchaseOrder.id.],
    [GraphQLError: Cannot return null for non-nullable field PurchaseOrder.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all schedule lines 1`] = `
Object {
  "data": Object {
    "allScheduleLines": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field ScheduleLine.id.],
    [GraphQLError: Cannot return null for non-nullable field ScheduleLine.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all supplier status 1`] = `
Object {
  "data": Object {
    "allSupplierStatus": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field SupplierStatus.id.],
    [GraphQLError: Cannot return null for non-nullable field SupplierStatus.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all suppliers 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field Supplier.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch all user 1`] = `
Object {
  "data": Object {
    "allUsers": Array [
      null,
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field User.id.],
    [GraphQLError: Cannot return null for non-nullable field User.id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one address 1`] = `
Object {
  "data": Object {
    "address": Object {
      "building_name": "building name",
      "city": "city",
      "id": "A1",
      "state": "state",
      "street": "street",
      "zip_code": "zip_code",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one item  1`] = `
Object {
  "data": Object {
    "item": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one purchase order 1`] = `
Object {
  "data": Object {
    "purchaseOrder": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one schedule line 1`] = `
Object {
  "data": Object {
    "scheduleLine": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one supplier 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field Query.supplier.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one supplier status 1`] = `
Object {
  "data": Object {
    "supplierStatus": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries should fetch one user 1`] = `
Object {
  "data": Object {
    "user": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries update a supplier 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [GraphQLError: Variable "$supplier" got invalid value { id: "1", supplierNo: "001", supplierName: "The Supplier", address: { building_name: "002", street: "Elmer", city: "Celadon", state: "Johto", zip_code: "123" }, contactPerson: "Basil Valdez", contactNumber: "1234567", tin: "12345" }; Field "address" is not defined by type UpdateSupplierInput.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries update a user 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [ValidationError: Variable "$user" of type "UpdateSupplierInput!" used in position expecting type "UpdateUserInput".],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
